src/controllers/howDoYouWantToPay.controller.ts

```diff
@@ -0,0 +1,145 @@
+      return this.redirect(SEARCH_COMPANY_URI)
+    }
+
+    if (dissolutionSession.paymentType) {
+      const body = {
+        paymentType: dissolutionSession.paymentType
+      } as HowDoYouWantToPayModel
+      return this.renderView(body)
+    }
+
+    return super.render('how-do-you-want-to-pay')
+  }
+
+  @httpPost('')
+  public async post(@requestBody() body: HowDoYouWantToPayModel): Promise<string | RedirectResult>  {
+    if (!this.PAY_BY_ACCOUNT_FEATURE_ENABLED) {
+      return this.redirect(SEARCH_COMPANY_URI)
+    }
+
+    const token: string = this.session.getAccessToken(this.httpContext.request)
+    const dissolutionSession: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
+
+    if (await this.isAlreadyPaid(dissolutionSession, token)) {
+      return this.redirect(SEARCH_COMPANY_URI)
+    }
+
+    const paymentType: PaymentType = body.paymentType
+
+    const errors: Optional<ValidationErrors> = this.validator.validate(body, howDoYouWantToPaySchema)
+    if (errors) {
+      return this.renderView(body, errors)
+    }
+
+    return this.redirect(await this.getRedirectURI(paymentType, dissolutionSession))
+  }
+
+  private async renderView(
+    data?: Optional<HowDoYouWantToPayModel>,
+    errors?: Optional<ValidationErrors>): Promise<string> {
+    const viewModel: ViewModel = {
+      data,
+      errors
+    }
+
+    return super.render('how-do-you-want-to-pay', viewModel, errors ? StatusCodes.BAD_REQUEST : StatusCodes.OK)
+  }
+
+  private async getRedirectURI(paymentType: PaymentType, dissolutionSession: DissolutionSession): Promise<string> {
+    const updatedSession = this.updateSessionPaymentType(dissolutionSession, paymentType)
+
+    if (paymentType === PaymentType.ACCOUNT) {
+      return PAY_BY_ACCOUNT_URI
+    }
+
+    return await this.getCreditCardPaymentUrl(updatedSession)
+  }
+
+  private async getCreditCardPaymentUrl(dissolutionSession: DissolutionSession): Promise<string> {
+    const token: string = this.session.getAccessToken(this.httpContext.request)
+
+    const paymentStateUUID: string = uuidv4()
+
+    const paymentURL: string = await this.paymentService.generatePaymentURL(token, dissolutionSession, paymentStateUUID)
+
+    this.updateSession(dissolutionSession, paymentStateUUID)
+
+    return paymentURL
+  }
+
+  private updateSession(dissolutionSession: DissolutionSession, paymentStateUUID: string): void {
+    const updatedSession: DissolutionSession = {
+      ...dissolutionSession,
+      paymentStateUUID
+    }
+
+    this.session.setDissolutionSession(this.httpContext.request, updatedSession)
+  }
+
+  private updateSessionPaymentType(dissolutionSession: DissolutionSession, paymentType: PaymentType): DissolutionSession {
+    const updatedSession: DissolutionSession = {
+      ...dissolutionSession,
+      paymentType
+    }
+
+    this.session.setDissolutionSession(this.httpContext.request, updatedSession)
```

Comment:
You're calling setDissolutionSession twice. Rework the code to ensure that it is only invoked once per request.